elide-async/src/main/java/com/yahoo/elide/async/models/AsyncQuery.java

```diff
@@ -41,48 +43,93 @@
-    @Max(10)
-    @ComputedAttribute
-    private Integer asyncAfterSeconds = 10;
+    @Override
+    public void setResult(AsyncAPIResult result) {
+        this.result = (AsyncQueryResult) result;
+    }
 
-    @Exclude
-    @Column(columnDefinition = "varchar(36)")
-    private String requestId = UUID.randomUUID().toString();
+    @Override
+    public AsyncAPIResult executeRequest(AsyncExecutorService service, User user, String apiVersion)
+            throws URISyntaxException, NoHttpResponseException {
+        ElideResponse response = null;
+        log.debug("AsyncQuery Object from request: {}", this);
+        if (queryType.equals(QueryType.JSONAPI_V1_0)) {
+            response = executeJsonApiRequest(service.getElide(), user, apiVersion);
+        } else if (queryType.equals(QueryType.GRAPHQL_V1_0)) {
+            response = executeGraphqlRequest(service.getRunners(), user, apiVersion);
+        }
+        nullResponseCheck(response);
+
+        AsyncQueryResult queryResult = new AsyncQueryResult();
+        queryResult.setHttpStatus(response.getResponseCode());
+        queryResult.setCompletedOn(new Date());
+        queryResult.setResponseBody(response.getBody());
+        queryResult.setContentLength(response.getBody().length());
+        //TODO Add recordcount to queryResultObj
+        return queryResult;
+    }
 
-    @UpdatePermission(expression = "(Principal is Admin OR Principal is Owner) AND value is Cancelled")
-    @CreatePermission(expression = "value is Queued")
-    @Enumerated(EnumType.STRING)
-    private QueryStatus status = QueryStatus.QUEUED;
+    /**
+     * This method parses the url and gets the query params.
+     * And adds them into a MultivaluedMap to be used by underlying Elide.get method
+     * @param query query from the Async request
+     * @throws URISyntaxException URISyntaxException from malformed or incorrect URI
+     * @return MultivaluedMap with query parameters
+     */
+    private MultivaluedMap<String, String> getQueryParams(String query) throws URISyntaxException {
+        URIBuilder uri;
+        uri = new URIBuilder(query);
+        MultivaluedMap<String, String> queryParams = new MultivaluedHashMap<String, String>();
+        for (NameValuePair queryParam : uri.getQueryParams()) {
+            queryParams.add(queryParam.getName(), queryParam.getValue());
+        }
+        return queryParams;
+    }
 
-    @Enumerated(EnumType.STRING)
-    @NotNull
-    private ResultType resultType = ResultType.EMBEDDED; //EMBEDDED, DOWNLOAD
+    /**
+     * This method parses the url and gets the query params.
+     * And retrieves path to be used by underlying Elide.get method
+     * @param query query from the Async request
+     * @throws URISyntaxException URISyntaxException from malformed or incorrect URI
+     * @return Path extracted from URI
+     */
+    private String getPath(String query) throws URISyntaxException {
+        URIBuilder uri;
+        uri = new URIBuilder(query);
```

Comment:
nitpick but we parse the query URL twice - once for the path and once for the headers.  We could combine these in a single parse.